### ISSUE TITLE
Publish as website (domains)

### ISSUE THREAD

**Issue by virgile-dev** [numerique-gouv/impress#522]

**Comment by virgile-dev** [comment]
Yowa needs this for rules as code.

**Comment by sampaccoud** [comment]
@virgile-dev I would say feasible but not trivial as it involves creating ingresses in Kubernetes on the fly. Anything easier than that @rouja ? 
It also raises the question of what domains we would allow...

**Comment by rouja** [comment]
It might be the simplest, but it’s not trivial. On-the-fly ingress is one example, while DNS and certificates are others.

**Comment by rl-83** [comment]
Proposition #2 : [FIGMA](https://www.figma.com/design/qdCWR4tTUr7vQSecEjCyqO/DOCS?node-id=6159-16992&t=cQua2H0k6HF3jXsR-1)

![Image](https://github.com/user-attachments/assets/cf38674e-a853-4d7f-abba-ca89cf8a52da)

**Comment by virgile-dev** [comment]
Hey @rl-83 should we add a view count somewhere ?
As a user I might to know how many people visited my knowledge base.

**Comment by rl-83** [comment]
@virgile-dev let's wait for user feedbacks to discuss the relevance of this feature?

**Comment by ethanc8** [comment]
Maybe for custom domains that already have their own web services, the export feature could be extended to provide a zip of HTML files that the user can upload to the web server? But then it would not update automatically.

I am also thinking that there could be a feature to push the generated HTML to a Git repo, which can be published on a Pages-like site or by having a script that periodically pulls the Git repo onto the web server.

This is probably for a different usecase than what this issue is talking about, however.

**Comment by sampaccoud** [comment]
Yeah that's smart. Docs would serve it on a path of its own domain and we would pull it from there to host it on a static site with a custom domain 💪

**Comment by i-am-unknown-81514525** [comment]
Any thoughts about this feature(document linking)? Or is this out of scope of the website idea:
A page can link to another document, where a User can add a document that they can view.
If you are in the editor and have the right to view/edit the other documents:
=> Redirect to the file with corresponding document 
If you are in the editor and **do not** have the right to view/edit the other documents, but the file were exported:
=> Redirect to page view of the corresponding document 
If you are in the page view and the other documents were exported 
=> Redirect to page view of the corresponding document 
Otherwise
=> Return a warning popup/redirect to non existing page that show 404

**Comment by virgile-dev** [comment]
Closed in favor of https://github.com/suitenumerique/docs/issues/1650
